src/NuGet.Clients/NuGet.CommandLine/Commands/Pack/AssemblyMetadataExtractor.cs

```diff
@@ -15,7 +15,7 @@ public static class AssemblyMetadataExtractor
     {
         private static T CreateInstance<T>(this AppDomain domain)
         {
-            return (T)domain.CreateInstanceAndUnwrap(typeof(T).Assembly.FullName, typeof(T).FullName);
+            return (T)domain.CreateInstanceFromAndUnwrap(Assembly.GetExecutingAssembly().Location, typeof(T).FullName);
```

Comment:
Please add comments why choose the CreateInstanceFromAndUnwrap instead of CreateInstanceAndUnwrap.

Reply:
Because it works when an assembly is renamed? :)

There are tests covering this scenario, and I don't see this as a complex change that requires additional explanation. 

My thought is adding a comment is too much, but can add it if we think it's necessary.